cmd/security-webhook/main.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+
+	configvalidation "github.com/google/knative-gcp/pkg/apis/configs/validation"
+	securityv1alpha1 "github.com/google/knative-gcp/pkg/apis/security/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/eventing/pkg/logconfig"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/injection/sharedmain"
+	"knative.dev/pkg/leaderelection"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/signals"
+	tracingconfig "knative.dev/pkg/tracing/config"
+	"knative.dev/pkg/webhook"
+	"knative.dev/pkg/webhook/certificates"
+	"knative.dev/pkg/webhook/configmaps"
+	"knative.dev/pkg/webhook/resourcesemantics"
+	"knative.dev/pkg/webhook/resourcesemantics/defaulting"
+	"knative.dev/pkg/webhook/resourcesemantics/validation"
+)
+
+var types = map[schema.GroupVersionKind]resourcesemantics.GenericCRD{
+	securityv1alpha1.SchemeGroupVersion.WithKind("HTTPPolicy"):         &securityv1alpha1.HTTPPolicy{},
+	securityv1alpha1.SchemeGroupVersion.WithKind("EventPolicy"):        &securityv1alpha1.EventPolicy{},
+	securityv1alpha1.SchemeGroupVersion.WithKind("HTTPPolicyBinding"):  &securityv1alpha1.HTTPPolicyBinding{},
+	securityv1alpha1.SchemeGroupVersion.WithKind("EventPolicyBinding"): &securityv1alpha1.EventPolicyBinding{},
+}
+
+func NewDefaultingAdmissionController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	// Decorate contexts with the current state of the config.
+	ctxFunc := func(ctx context.Context) context.Context {
+		return ctx
+	}
+
+	return defaulting.NewAdmissionController(ctx,
+
+		// Name of the default webhook.
+		"webhook.security.knative.dev",
```

Comment:
It's in a weird state now so I'm open to suggestions. Here is a little background.

Security policy started as something for Eventing but gradually evolved to have a bigger scope - between security boundaries events are transported with http; to provide events policies problem essentially we would need to rely on http policies; and http policy is generic. So in the design we proposed the feature track to be knative-scope rather than eventing-scope. That's why I'm putting `security.knative.dev` here and as the API group name.

However, the progress of the feature track has been really slow in the community. So we decided to put it in knative-gcp first so that we can at least start coding and provide it for google users. That's why the code ended up in knative-gcp.

I struggled with the naming because ideally it would be `security.knative.dev` but it's not something knative officially supports today. Naming it with `events.cloud.google.com` also seems not appropriate because it's not meant to be just for eventing.

Reply:
I don't know how we are allowed to use the `knative.dev` namespace if this isn't within the knative project. I would lean towards something else, especially if we know we want to change it later anyway (e.g. to remove `experimental`). How about `security.experimental.events.cloud.google.com`? My reasoning being that we are allowed to use `events.cloud.google.com`. I agree that it implies that this security is only for eventing, but as we intend to change the name in the future anyway, I don't see that as a problem while we are experimenting.

Reply:
Track in https://github.com/google/knative-gcp/issues/726

Reply:
one extra thing... we are using `messaging.cloud.google.com` and `pubsub.cloud.google.com` for Channel, and Topic/PullSubscription respectively... Based on @Harwayne 's point, I think we also need to change it to `messaging.events.cloud.google.com` and `pubsub.events.cloud.google.com`... 
@yolocs if you loop in Ron, can you ask about this as well? We rather do the whole renaming now in v1alpha1 before we move all to v1beta1. 
And I think I'd vote for `security.events.cloud.google.com`, without the `experimental`, but no strong opinions.